cmake/modules/AddSwift.cmake

```diff
@@ -854,6 +848,31 @@ function(_add_swift_library_single target name)
     set(SWIFTLIB_SINGLE_API_NOTES "${module_name}")
   endif()
 
+  # On Linux and FreeBSD add conformance magic objects that coalesce protocol
+  # conformance sections. This wouldn't be necessary if the link was done by
+  # the swift binary: rdar://problem/19007002
+  if("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux" OR
+     "${CMAKE_SYSTEM_NAME}" STREQUAL "FreeBSD")
+
+    if("${libkind}" STREQUAL "SHARED")
+      set(arch_subdir "${SWIFTLIB_DIR}/${SWIFTLIB_SINGLE_SUBDIR}")
+
+      add_custom_command(
+        OUTPUT  "${CMAKE_CURRENT_BINARY_DIR}/swift_begin.o"
+        COMMAND "${CMAKE_COMMAND}" -E copy "${arch_subdir}/swift_begin.o" "${CMAKE_CURRENT_BINARY_DIR}/swift_begin.o"
```

Comment:
Why do we need to copy object files around?  Is this working around a limitation of CMake?